### PR TITLE
Add loop counter logic and AnimatableImageView delegate

### PR DESCRIPTION
**Pull request by VitorMakoto**
The main loop logic was extracted from
kaishin#42, a open PR from @storix
that was still not merged at the time this commit was created

Add loop counter logic and optional delegate method which is called when
an animation stops

Add a public getter for the current loop count and a setter/getter for
the max number of loops. If max number of loops is set to 0 it loops
forever

Add loadedData which hold the data that is currently loaded in the
AnimatableImageView. It has a private setter and a public getter

### CODE REVIEW

**Comment by igortoliveira** [comment]
LGTM

**Comment by VitorMakoto** [comment]
Updated

File: Source/AnimatableImageView.swift [hunk 1]
```diff
@@ -1,18 +1,42 @@
...
+
+public extension AnimatableImageViewDelegate {
+  /// forces method to be optional
+  func animatableImageViewAnimationDidStop(animatableImageView: AnimatableImageView) { }
+}
+
 /// A subclass of `UIImageView` that can be animated using an image name string or raw data.
 public class AnimatableImageView: UIImageView {
+  /// animatable image view delegate instance
+  public weak var delegate : AnimatableImageViewDelegate?
```
Comment (igortoliveira) [inline Source/AnimatableImageView.swift]: space before `:`

File: Source/AnimatableImageView.swift [hunk 2]
```diff
@@ -75,13 +102,44 @@ public class AnimatableImageView: UIImageView {
...
+
+        strongSelf.delegate?.animatableImageViewAnimationDidStop(strongSelf)
+      }
+    }
+  }
+
+  /// Specifies whether all of the required animation iterations have been finished.
+  /// - returns: true if the animation should stop looping, false otherwise.
+  func shouldStopLooping() -> Bool {
+    guard maxLoopCount > 0 else {
```
Comment (igortoliveira) [inline Source/AnimatableImageView.swift]: `return maxLoopCount > 0 && maxLoopCount == currentLoopCount` is enough,

**Pull request merged**